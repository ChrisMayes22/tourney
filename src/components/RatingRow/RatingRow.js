import React, { Component } from 'react';
import RatingButton from '../../components/RatingButton/RatingButton';
import classes from './RatingRow.css';

export class RatingRow extends Component {
    state = {
        qualityArray: [ {quality: 'awful', active: false, points: 1}, 
                        {quality: 'poor', active: false, points: 2}, 
                        {quality: 'average', active: false, points: 3}, 
                        {quality: 'good', active: false, points: 4}, 
                        {quality: 'excellent', active: false, points: 5}]
    }

    ratingChoiceHandler = (id) =>{
        const qualityArrCopy = [{...this.state.qualityArray[0]},
            {...this.state.qualityArray[1]},
            {...this.state.qualityArray[2]},
            {...this.state.qualityArray[3]},
            {...this.state.qualityArray[4]}];
        
        const transformedCopy = qualityArrCopy.map((el, i) => {
            if(i <= id){
                el.active = true;
                return el;
            } else {
                el.active = false;
                return el;
            }
        })

        this.setState({
            qualityArray: transformedCopy
        })
    }

    ratingRowManager = (id, points, rowId, character) =>{
        this.ratingChoiceHandler(id);
        this.props.clicked(points, rowId, character);
    }

    render(){
        return(
            <div id={this.props.id}
                className={this.props.players === 3 ? classes.flexContainer__row3Players : classes.flexContainer__row4Players}>
                    {this.state.qualityArray.map((el, i) => {
                        let isChosen = '';
                        if(this.state.qualityArray[i].active){
                            isChosen = 'active';
                        }
                        return(
                            <RatingButton children={i + 1}
                                quality={classes[el.quality]}
                                chosen={classes[isChosen]}
                                key={i}
                                clicked={() => this.ratingRowManager(i, el.points, this.props.rowId, this.props.character)}
                                />
                        );
                    })}
                </div>
        );
    }
}



export default RatingRow;