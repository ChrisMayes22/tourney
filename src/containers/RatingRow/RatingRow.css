.flexContainer__row3Players{
    display: flex;
    flex-direction: row;
    justify-content: center;
    min-height: 5vh;
    max-width: 75%;
    margin: 3px;
    border: black solid;
}

.flexContainer__row4Players{
    display: flex;
    flex-direction: row;
    justify-content: center;
    min-height: 5vh;
    max-width: 95%;
    margin: 3px;
    border: black solid;
}

.awful{
    background-color: rgba(151, 12, 23);
}

.poor{
    background-color: rgba(151, 12, 23, 0.1);
}

.poor:hover,
.active.poor{
    background-color: rgba(151, 12, 23);
}


.average{
    background-color: rgba(151, 12, 23, 0.1);
}

.average:hover,
.active.average{
    background-color: rgba(151, 12, 23);
}

.good{
    background-color: rgba(151, 12, 23, 0.1);
}

.good:hover,
.active.good{
    background-color: rgba(151, 12, 23);
}

.excellent{
    background-color: rgba(151, 12, 23, 0.1);
}

.excellent:hover,
.active.excellent{
    background-image: url('../../assets/background__flame-on.jpg');
    background-size: cover;
}