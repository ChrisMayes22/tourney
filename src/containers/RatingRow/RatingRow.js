import React, { Component } from 'react';
import RatingButton from '../../components/RatingButton/RatingButton';
import classes from './RatingRow.css';
import { connect } from 'react-redux';
import * as actionTypes from '../../reducers/actions';

class RatingRow extends Component {
    state = {
        qualityArray: [ {quality: 'awful', active: false, points: 1}, 
                        {quality: 'poor', active: false, points: 2}, 
                        {quality: 'average', active: false, points: 3}, 
                        {quality: 'good', active: false, points: 4}, 
                        {quality: 'excellent', active: false, points: 5}]
    }

    ratingChoiceHandler = (id) =>{
        const qualityArray = JSON.parse(JSON.stringify(this.state.qualityArray));
        
        qualityArray.forEach(el => {
            const currentIndex = qualityArray.indexOf(el);
            if(currentIndex <= id){
                el.active = true;
            } else {
                el.active = false;
            }
        })
        this.setState({
            qualityArray: qualityArray
        })
    }

    ratingRowManager = (id, points, rowId, character) =>{
        this.ratingChoiceHandler(id);
        this.props.onRate(points, rowId, character);
    }

    render(){
        return(
            <div className={this.props.players == 3 ? classes.flexContainer__row3Players : classes.flexContainer__row4Players}>
                    {this.state.qualityArray.map(el => {
                        const qualityArray = [...this.state.qualityArray]
                        const currentIndex = qualityArray.indexOf(el);
                        let isChosen = '';
                        if(qualityArray[currentIndex].active){
                            isChosen = 'active';
                        }
                        return(
                            <RatingButton children={currentIndex + 1}
                                quality={classes[el.quality]}
                                chosen={classes[isChosen]}
                                key={currentIndex}
                                clicked={() => this.ratingRowManager(currentIndex, el.points, this.props.rowId, this.props.character)}
                                />
                        );
                    })}
                </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        players: state.players
    }
};

const mapDispatchToProps = dispatch => {
    return {
        onRate: (points, rowId, character) => {
            dispatch({type: actionTypes.RATE_CHARACTER, payload: {points: points, rowId: rowId, character: character}})
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(RatingRow);