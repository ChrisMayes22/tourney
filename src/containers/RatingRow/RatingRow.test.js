import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { shallow } from 'enzyme';

import RatingRow from './RatingRow';

const wrapper = shallow(<RatingRow/>)

describe('Basic render tests for component', function(){
    test('App renders as expected', () => {
      
      expect(wrapper).toMatchSnapshot();
    })
})