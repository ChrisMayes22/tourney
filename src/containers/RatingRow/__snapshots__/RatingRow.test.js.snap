// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When unconnected component first renders RatingRow renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RatingRow
    character={
      Object {
        "id": "foo$id3",
      }
    }
    players={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ratingButton
          clicked={[Function]}
        >
          1
        </ratingButton>,
        <ratingButton
          clicked={[Function]}
        >
          2
        </ratingButton>,
        <ratingButton
          clicked={[Function]}
        >
          3
        </ratingButton>,
        <ratingButton
          clicked={[Function]}
        >
          4
        </ratingButton>,
        <ratingButton
          clicked={[Function]}
        >
          5
        </ratingButton>,
      ],
      "className": undefined,
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": 1,
          "chosen": undefined,
          "clicked": [Function],
          "quality": undefined,
        },
        "ref": null,
        "rendered": 1,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": 2,
          "chosen": undefined,
          "clicked": [Function],
          "quality": undefined,
        },
        "ref": null,
        "rendered": 2,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "children": 3,
          "chosen": undefined,
          "clicked": [Function],
          "quality": undefined,
        },
        "ref": null,
        "rendered": 3,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "children": 4,
          "chosen": undefined,
          "clicked": [Function],
          "quality": undefined,
        },
        "ref": null,
        "rendered": 4,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "children": 5,
          "chosen": undefined,
          "clicked": [Function],
          "quality": undefined,
        },
        "ref": null,
        "rendered": 5,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ratingButton
            clicked={[Function]}
          >
            1
          </ratingButton>,
          <ratingButton
            clicked={[Function]}
          >
            2
          </ratingButton>,
          <ratingButton
            clicked={[Function]}
          >
            3
          </ratingButton>,
          <ratingButton
            clicked={[Function]}
          >
            4
          </ratingButton>,
          <ratingButton
            clicked={[Function]}
          >
            5
          </ratingButton>,
        ],
        "className": undefined,
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": 1,
            "chosen": undefined,
            "clicked": [Function],
            "quality": undefined,
          },
          "ref": null,
          "rendered": 1,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": 2,
            "chosen": undefined,
            "clicked": [Function],
            "quality": undefined,
          },
          "ref": null,
          "rendered": 2,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "children": 3,
            "chosen": undefined,
            "clicked": [Function],
            "quality": undefined,
          },
          "ref": null,
          "rendered": 3,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "children": 4,
            "chosen": undefined,
            "clicked": [Function],
            "quality": undefined,
          },
          "ref": null,
          "rendered": 4,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "children": 5,
            "chosen": undefined,
            "clicked": [Function],
            "quality": undefined,
          },
          "ref": null,
          "rendered": 5,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
